File: easy/Lettercase-Percentage-Ratio/Lettercase-Percentage-Ratio.java
import java.io.*;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            ShowPercentage(line);
        }
    }
    
    static void ShowPercentage(String line){
        int leng = line.length();
        int upper = 0;
        int lower = 0;
        for(int i=0;i<leng;i++){
            String ch = line.substring(i,i+1);
            if(ch.matches("[A-Z]")) upper++;
            else lower++;
        }
        System.out.printf("lowercase: %.2f", (float)lower*100/leng);
        System.out.printf(" uppercase: %.2f", (float)upper*100/leng);
        System.out.println();
    }
}

File: easy/Lettercase-Percentage-Ratio/Lettercase-Percentage-Ratio.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Text.RegularExpressions;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            ShowPercentage(line);
        }
    }
    
    static void ShowPercentage(string line){
        int leng = line.Length;
        int upper = 0;
        int lower = 0;
        Regex rgx = new Regex(@"[A-Z]");
        for(int i=0;i<leng;i++){
            if(rgx.IsMatch(line.Substring(i,1))) upper++;
        }
        decimal upperporc = (decimal) upper*100/ (decimal) leng;
        Console.WriteLine($"lowercase: {100 - upperporc:N2} uppercase: {upperporc:N2}");
    }
}
